### DISCUSSION TITLE
Play from iCloud

### DISCUSSION THREAD

**Discussion by bsrobinson** [TortugaPower/BookPlayer#608] [Ideas]
I have a large number of Audiobook files, too many to store on my phone.  I would really like to browse and play the books directly from iCloud without having to import or store them locally (obv. books I'm listening to would be cached)

Ideally I think there would be a 'BookPlayer' folder in iCloud where I could create folders of books (similar to how iTunes would have stored them); I should then be able browse the library in that folder (maybe after an index) and play from there.

Is this possible / something you would be interested in adding?

Many thanks,

Ben

**Comment by pichfl** [comment]
This isn't quite possible as this would violate the sandboxing in iOS as far as I understand. Simply put: BookPlayer can't access iCloud that way.

**Comment by bsrobinson** [comment]
Hi,

Thanks for the reply.

Is that really true though - I’m not an expert on this, so happy to be incorrect - however I have an app installed on my iPhone called ‘Arc’; it has created a folder at the root of iCloud Drive (or at least that where it appears); and that app can freely read and write to/from that folder - I assume this means it is part of that apps sandbox?

So you wouldn’t have a free run over my iCloud Drive, I would have to move my audiobooks into your sandbox, but that would be straight forward as a user…

Like I say, happy to be corrected if that’s incorrect :)

Thanks, Ben

**Reply by pichfl** [reply]
That would indeed be possible. Only folders created by an app can be controlled by it.

Given the new folder support we just released, we can look into options. However don't hold your breath for a quick implementation of this, we generally do not want to implement vendor specific solutions and that includes Apple as a cloud file provider. 

Don't take this the wrong way, I do like the idea 👍

**Comment by bsrobinson** [comment]
I might argue that the native cloud solution isn’t any more vendor specific that the app itself 😁  Thanks for thinking about it though 👍

I’ve decided this is the app I will use going forward, so I’ll download the code and try and get into it; wish me luck!

Ben.

**Comment by LordZork** [comment]
@pichfl and @GianniCarlo,

I was just about to post a similar idea, but saw this one here.  I would also love this feature.  The only difference (with my idea) would be the inclusion of other cloud services and/or servers (like the [Plex](https://www.plex.tv/) [Media Server](https://www.plex.tv/media-server-downloads/#plex-media-server)), not just iCloud.

Though, the main idea is the same—the ability to simply stream (and maintain bookmarks, location, and completion) from a cloud drive or server, instead of having to import or have the actual files on the iPhone.

That said, and I'm no expert here, but I think @hc98br has a good point.  The native iCloud solution shouldn't be any more vendor specific than the app itself... and it would be amazing!

Integrating other cloud services and/or media servers would simply be a bonus.

**Comment by darren365** [comment]
The idea of the Library not being restricted to on device books is exactly what I'm looking for. Where the Library can be browsed on device and where a book is to be played, it is downloaded when requested by the user.
